File: main/src/main/java/com/example/main/data/source/remote/DataSourceMainRepository.java
package com.example.main.data.source.remote;

import com.example.core.data.model.Resource;
import com.example.core.data.model.movie.Movie;
import com.example.core.util.SingleLiveEvent;

public interface DataSourceMainRepository {

    void getMovies(SingleLiveEvent<Resource<Movie>> movieSingleLiveEvent, Integer page);
    void getTopMovies(SingleLiveEvent<Resource<Movie>> movieTopSingleLiveEvent, Integer page);
}

File: main/src/main/java/com/example/main/data/source/remote/DataSourceMainRepository.kt
package com.example.main.data.source.remote

import com.example.core.data.model.Resource
import com.example.core.data.model.movie.Movie
import com.example.core.util.SingleLiveEvent

interface DataSourceMainRepository {

    fun getMovies(movieSingleLiveEvent: SingleLiveEvent<Resource<Movie>>, page: Int)
    fun getTopMovies(movieTopSingleLiveEvent: SingleLiveEvent<Resource<Movie>>, page: Int)
}
